services/core-api/app/api/now_applications/resources/now_application_import_resource.py

```diff
@@ -58,6 +58,9 @@ def post(self, application_guid):
         application.latitude = latitude
         application.longitude = longitude
         application.now_application_guid = application_guid
+        db.session.refresh(now_application_identity)
+        if now_application_identity.now_application_id is not None:
+            raise BadRequest('This redcord has already been imported.')
         application.save()
```

Comment:
can we leave a note that this is not perfect due to the request processing time and multiple requests fired at once. 

Great first cut to avoid issues.

Reply:
redcord --> record